src/org/exist/collections/Collection.java

```diff
@@ -1096,6 +1096,20 @@ public void removeCollection(final DBBroker broker, final XmldbURI name) throws
         }
     }
 
+    public void removeResource(Txn txn, DBBroker broker, DocumentImpl doc) throws PermissionDeniedException, LockException, IOException, TriggerException {
+        if (doc == null) return;
+
+        if (doc.getCollection() != this) {
```

Comment:
Sorry I don't understand your response - How is `DocumentImpl` not already outside this method? You have to pass it in to this method!

Inside eXist itself this should never happen, if someone is calling this and has that problem in their app, they have a broken app and should solve that there... no?